sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/test/scala/com/azure/cosmos/spark/CosmosCatalogSpec.scala

```diff
@@ -3,160 +3,94 @@
-
-    val databaseName = RandomStringUtils.randomAlphabetic(5).toLowerCase
+    val databaseName = getAutoCleanableDatabaseName()
     val containerName = RandomStringUtils.randomAlphabetic(6).toLowerCase + System.currentTimeMillis()
+    cleanupDatabaseLater(databaseName)
 
-    spark.sql(s"CREATE DATABASE cosmoscatalog.${databaseName};")
-    spark.sql(s"CREATE TABLE cosmoscatalog.${databaseName}.${containerName} (word STRING, number INT) using cosmos.items;")
+    spark.sql(s"CREATE DATABASE testCatalog.${databaseName};")
+    spark.sql(s"CREATE TABLE testCatalog.${databaseName}.${containerName} (word STRING, number INT) using cosmos.items;")
 
-    val containerProperties = client.getDatabase(databaseName).getContainer(containerName).read().block().getProperties
+    val containerProperties = cosmosClient.getDatabase(databaseName).getContainer(containerName).read().block().getProperties
 
     // verify default partition key path is used
     containerProperties.getPartitionKeyDefinition.getPaths.asScala.toArray should equal(Array("/id"))
 
     // validate throughput
-
-    val throughput = client.getDatabase(databaseName).getContainer(containerName).readThroughput().block().getProperties
+    val throughput = cosmosClient.getDatabase(databaseName).getContainer(containerName).readThroughput().block().getProperties
     throughput.getManualThroughput shouldEqual 400
-
-    client.close()
-    spark.close()
   }
 
-  it should "create a table with customized properties" taggedAs (RequiresCosmosEndpoint) in {
-    val cosmosEndpoint = TestConfigurations.HOST
-    val cosmosMasterKey = TestConfigurations.MASTER_KEY
-
-    val client = new CosmosClientBuilder()
-      .endpoint(cosmosEndpoint)
-      .key(cosmosMasterKey)
-      .buildAsyncClient()
-
-    val spark = SparkSession.builder()
-      .appName("spark connector sample")
-      .master("local")
-      .enableHiveSupport()
-      .getOrCreate()
-
-    spark.conf.set(s"spark.sql.catalog.cosmoscatalog", "com.azure.cosmos.spark.CosmosCatalog")
-    spark.conf.set(s"spark.sql.catalog.cosmoscatalog.spark.cosmos.accountEndpoint", cosmosEndpoint)
-    spark.conf.set(s"spark.sql.catalog.cosmoscatalog.spark.cosmos.accountKey", cosmosMasterKey)
-
-    val databaseName = RandomStringUtils.randomAlphabetic(5).toLowerCase
+  it can "create a table with customized properties" taggedAs (RequiresCosmosEndpoint) in {
+    val databaseName = getAutoCleanableDatabaseName()
     val containerName = RandomStringUtils.randomAlphabetic(6).toLowerCase + System.currentTimeMillis()
 
-    spark.sql(s"CREATE DATABASE cosmoscatalog.${databaseName};")
-    spark.sql(s"CREATE TABLE cosmoscatalog.${databaseName}.${containerName} (word STRING, number INT) using cosmos.items " +
+    spark.sql(s"CREATE DATABASE testCatalog.${databaseName};")
+    spark.sql(s"CREATE TABLE testCatalog.${databaseName}.${containerName} (word STRING, number INT) using cosmos.items " +
```

Comment:
Unrelated - but I really love that you have wired up the metadata changes already - I am sure Spark developers will love this feature - don't know how many customers I have seen that had taken a dependency to some crappy very old Python SDK just to be able to create/modify containers etc.